django/db/migrations/state.py

```diff
@@ -258,7 +298,19 @@ def rename_field(self, app_label, model_name, old_name, new_name):
                         new_name if to_field_name == old_name else to_field_name
                         for to_field_name in to_fields
                     ])
-        self.reload_model(app_label, model_name, delay=delay)
+        if self._relations is not None:
+            old_name_lower = old_name.lower()
+            new_name_lower = new_name.lower()
+            for from_model, to_model in self._relations.items():
+                # It's safe to modify the same collection that is interated
```

Comment:
```suggestion
                # It's safe to modify the same collection that is iterated
```

Reply:
Something else I noticed above there is that `from_model` isn't used, so it can be `values()`. (Sorry, @felixxm, for adding so many minor comments!)

Reply:
Updated